lib/spack/llnl/util/tty/__init__.py

```diff
@@ -65,20 +134,37 @@ def process_stacktrace(countback):
     return st_text
 
 
+def get_debug_text():
+    """When debugging is enabled, return a formatted string with milliseconds
+    since epoch"""
+    if _debug:
```

Comment:
I like that the timestamps are printed for *all* message types when `debug` is enabled.  Can you add one more tweak?  Make it so that you can control timestamps separately from debug.  We might want, for example, to print timestamps for regular `tty.msg` calls without having the debug output.  Put another way, `debug` should imply timestamps, but timestamps doesn't have to imply `debug`.

Reply:
I've added ```set_timestamp()``` function, although I haven't wired it to a command line argument.

Reply:
Added --timestamp flag